proto/gds/global/v1/global.proto

```diff
@@ -0,0 +1,94 @@
+syntax = "proto3";
+
+package gds.global.v1;
+option go_package = "github.com/trisacrypto/directory/pkg/gds/global/v1;global";
+
+import "gds/peers/v1/peers.proto";
+import "gds/models/v1/models.proto";
+import "trisa/gds/models/v1beta1/models.proto";
+
+// The Replication Service implements a gossip-based anti-entropy protocol for
+// synchronizing GDS replicas around the world. GDS expects total replication, meaning
+// all replicas will have a copy of all replicated data (although some data won't be
+// replicated). Not only is the Replication service is the mechanism by which data is
+// exchanged, but also network membership and discovery. Every replica should
+// periodically perform anti-entropy via the Gossip RPC with a randomly selected peer,
+// and update its data store via a "latest writer wins"policy.
+//
+// The Replication Service requires mTLS authentication in order to conduct exchanges.
+service Replication {
+    // During gossip, the initiating replica sends a randomly selected remote peer the
+    // version vectors of all objects it currently stores. The remote peer should
+    // respond with updates that correspond to more recent versions of the objects. The
+    // remote peer can than also make a reciprocal request for updates by sending the
+    // set of versions requested that were more recent on the initiating replica, and
+    // use a partial flag to indicate that it is requesting specific versions. This
+    // mechanism implements bilateral anti-entropy: a push and pull gossip.
+    rpc Gossip(VersionVectors) returns (Updates) {}
+}
+
+// VersionVectors initiate a Gossip RPC by passing along the current state of the
+// namespace of objects on the initiating replica. Objects should not have data
+// populated in this message, only metadata. If the partial flag is set, that indicates
+// that the replica only wants to consider the objects described rather than the entire
+// namespace (e.g. objects that may have been created outside the set of objects
+// described). The partial flag is typically used as a mechanism to fetch specific
+// objects that are known to be later from the remote replica. The namespaces array
+// specifies which namespaces should be considered in gossip allowing for partial
+// replication by namespace if necessary. If the namespaces array is empty, then all
+// namespaces are considered during Gossip.
+message VersionVectors {
+    // Version vectors of objects without data.
+    repeated Object objects = 1;
+
+    // A request to consider only the objects specified in the version vector and not
+    // the entire namespace (e.g. ignore objects that are not specified).
+    bool partial = 2;
+
+    // Limit the comparison only to the specified namespace(s). If not set or empty,
+    // then all namespaces are considered.
+    repeated string namespaces = 3;
```

Comment:
Added this field which allows for partial replication, but may also simplify things; e.g. if we just want to deal with Peers.

Reply:
Makes sense to me &mdash; probably a lot of partial updates will be at the level of granularity of namespaces, so I agree that this should simplify things